server/src/api/observation/observation.controller.ts

```diff
@@ -18,18 +18,20 @@ class ObservationController {
   async createObservation(req: ApiRequest<EdipiParam, ObservationApiModel>, res: Response) {
     Log.info('Creating observation', req.body);
 
-    const { reportingGroup, reportSchemaId, edipi, timestamp } = assertRequestBody(req, [
+    const { reportingGroup, reportSchemaId, edipi, timestamp, phoneNumber } = assertRequestBody(req, [
       'reportSchemaId',
       'edipi',
       'timestamp',
+      'phoneNumber',
     ]);
 
-    const rosters: RosterInfo[] = await getRostersForIndividual(edipi, `${timestamp}`);
+    const rosters: RosterInfo[] = await getRosterInfosForIndividualOnDate(edipi, `${timestamp}`);
 
     for (const roster of rosters) {
       if (hasReportingGroup(roster, reportingGroup)) {
         const reportSchema = await findReportSchema(reportSchemaId, roster.unit.org?.id);
         if (reportSchema) {
+          await rosterController.savePhoneNumber(edipi, phoneNumber);
           return res.json(saveObservationWithReportSchema(req, reportSchema));
```

Comment:
It just occurred to me that this chunk should be wrapped in a transaction, since there are multiple database updates happening.

If you search "transaction" in the project you should see several examples of how we're doing those. You'll probably want to add a `manager` argument to the `savePhoneNumber` function as well and use that to save so that it stays within the transaction.

Reply:
done